From f088cd12c62d30799535d1b9e3b2a54a180415a6 Mon Sep 17 00:00:00 2001
From: Masahiro Yamada <masahiroy@kernel.org>
Date: Mon, 26 Sep 2022 18:02:29 +0900
Subject: [PATCH 32/38] kallsyms: ignore __kstrtab_* and __kstrtabns_* symbols

Every EXPORT_SYMBOL creates __kstrtab_* and __kstrtabns_*, which
consumes 15-20% of the kallsyms entries.

For example, on the system built from the x86_64 defconfig,

  $ cat /proc/kallsyms | wc
     129527    388581   5685465
  $ cat /proc/kallsyms | grep __kstrtab | wc
      23489     70467   1187932

We already ignore __crc_* symbols populated by EXPORT_SYMBOL, so it
should be fine to ignore __kstrtab_* and __kstrtabns_* as well.

This makes vmlinux a bit smaller.

  $ size vmlinux.before vmlinux.after
     text    data     bss     dec     hex filename
  22785374        8559694 1413328 32758396        1f3da7c vmlinux.before
  22785374        8137806 1413328 32336508        1ed6a7c vmlinux.after

Signed-off-by: Masahiro Yamada <masahiroy@kernel.org>
---
 scripts/mksysmap | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/scripts/mksysmap b/scripts/mksysmap
index 75f3dfd1c..16a08b8ef 100755
--- a/scripts/mksysmap
+++ b/scripts/mksysmap
@@ -43,6 +43,8 @@
 #  $                    - local symbols for ARM, MIPS, etc.
 #  .L                   - local labels, .LBB,.Ltmpxxx,.L__unnamed_xx,.LASANPC, etc.
 #  __crc_               - modversions
+#  __kstrtab_           - EXPORT_SYMBOL (symbol name)
+#  __kstrtabns_         - EXPORT_SYMBOL (namespace)
 #
 # Ignored symbols:
 #  L0                   - for LoongArch?
@@ -52,5 +54,7 @@ $NM -n $1 | grep -v		\
 	-e ' \$'		\
 	-e ' \.L'		\
 	-e ' __crc_'		\
+	-e ' __kstrtab_'	\
+	-e ' __kstrtabns_'	\
 	-e ' L0$'		\
 > $2
-- 
2.38.0.rc1.6.g4fd6c5e444

